File: src/main/java/com/sica/backend/repository/UserRepository.java
package com.sica.backend.repository;

import com.sica.backend.entity.Role;
import com.sica.backend.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
    Optional<User> findByUsername(String username);
    List<User> findByRole(Role role);
    List<User> findByActiveTrue();
    Boolean existsByUsername(String username);
}

File: src/main/kotlin/com/sica/backend/repository/UserRepository.kt
package com.sica.backend.repository

import com.sica.backend.entity.Role
import com.sica.backend.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, String> {
    fun findByUsername(username: String): Optional<User>
    fun findByRole(role: Role): List<User>
    fun findByActiveTrue(): List<User>
    fun existsByUsername(username: String): Boolean
}
